package.json

```diff
@@ -18,7 +18,42 @@
     "karma-jasmine": "~2.0.1"
   },
   "scripts": {
-    "postinstall": "bash tools/scripts/npm/postinstall.sh",
+    "postinstall": "bash tools/scripts/npm/postinstall.sh && node -e \"try { require('fs').symlinkSync(require('path').resolve('node_modules/@bower_components'), 'bower_components', 'junction') } catch (e) { }\"",
```

Comment:
This is an interesting bit. Having the symlink addresses concerns about broken URLs, so it seems like a nice transition mechanism. But symlinks are not universal; for example:

* `tar` (as in `civicrm-X.Y.Z-drupal.tar.gz`, `civicrm-X.Y.Z-backdrop.tar.gz`) and `git` support symlinks
* `zip` (as in `civicrm-X.Y.Z-wordpress.zip`, `civicrm-X.Y.Z-joomla.zip`) does not support symlinks. (If you make some test content with symlinks; put it in a zip; and extract -- then you wind up with multiple copies of the files.)
* POSIX style file systems (ext2/ext3/ext4; HFS+) support symlinks
* Traditional Windows file systems (FAT) do not support symlinks.
* Network file systems are a wildcard. (NFS on a POSIX OS is fine; but the web-hosting world has some folks using weirder filesystems.)

There are a variety of workflows/environments in the ecosystem. Even if symlinks work at the two end-points (e.g. the `civicrm.org` build server and the downstream production server), the workflow can still get foobar'd if one component in the path doesn't support them. (Ex: `c.o` publishes a tarball. An implementer downloads it to his Windows workstation. Then he commits to the org's mono-repo. Then the prod server updates from the mono-repo.)

So the rub is:

* If we want to emphasize compatibility in file/URL structure, then symlinks are good.
* If we want to emphasize compatibility in workflows/tooling, then symlinks are bad.
* If we want to emphasize upfront support costs, then symlinks are good.
* If we want to emphasize future support costs, then symlinks are bad.

I haven't formed a full opinion yet on this PR/approach, but (for sake of argument)... if we want to do some kind of transitional symlink, then it might make sense to keep the symlinks out of the official build/release systems and to compensate with status-check. To wit:

```
if ("select * from civicrm_log" shows that this systems originated with an older version)
  and if (we're not on windows)
     and if (there is no bower_components symlink; or if the symlink is malformed)
       and if (we're in some transitional period -- defined by date or version#)
         then the status-check should show a tip to add a transitional symlink
```

Reply:
My inclination is not to try to use symlinks. It adds 'another thing that can go wrong', for the reasons you say above.